<!DOCTYPE html>

<html lang="en">

<head>
  
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-R5FPWGD7WK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-R5FPWGD7WK');
  </script>
	
	<title>Charles Rocabert | Research</title>
	
	<link rel="apple-touch-icon" sizes="180x180" href="img/favicon.png">
	<link rel="icon" type="image/png" sizes="32x32" href="img/favicon.png">
	<link rel="icon" type="image/png" sizes="16x16" href="img/favicon.png">
	
	<meta charset="utf-8" />
	<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
	
  <link rel="stylesheet" href="css/bootstrap-4.2.1.min.css">
  <link rel="stylesheet" href="../css/bootstrap-4.2.1.min.css">
  <link rel="stylesheet" href="../../css/bootstrap-4.2.1.min.css">
	<link rel="stylesheet" href="css/fontawesome-5.6.3.min.css">
  <link rel="stylesheet" href="../css/fontawesome-5.6.3.min.css">
  <link rel="stylesheet" href="../../css/fontawesome-5.6.3.min.css">

  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css" integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS" crossorigin="anonymous">
	<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">

  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.2/dist/katex.min.css" integrity="sha384-MlJdn/WNKDGXveldHDdyRP1R4CTHr3FeuDNfhsLPYrq2t0UBkUdK2jyTnXPEK1NQ" crossorigin="anonymous">

  
  <script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.2/dist/katex.min.js" integrity="sha384-VQ8d8WVFw0yHhCk5E8I86oOhv48xLpnDZx5T9GogA/Y84DcCKWXDmSDfn13bzFZY" crossorigin="anonymous"></script>

  
  <script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.2/dist/contrib/auto-render.min.js" integrity="sha384-+XBljXPPiv+OzfbB3cVmLHf4hdUFHlWNZN5spNQ7rmHTXpd7WvJum6fIACpNNfIR" crossorigin="anonymous"
      onload="renderMathInElement(document.body);"></script>
  <script>
  document.addEventListener("DOMContentLoaded", function()
  {
    renderMathInElement(document.body,
    {
      delimiters: [
        {left: "$$", right: "$$", display: true},
        {left: "$", right: "$", display: false}
      ]
    });
  });
  </script>

	<style>
	html {
		font-size: 1rem;
	}
	a {
		color: rgb(64,139,179);
		font-weight: normal;
	}
	.active a {
		color: rgb(26,115,232) !important;
		font-weight: bold;
		border-bottom: 5px solid !important;
		border-bottom-color: rgb(26,115,232);
	}
	.btn-outline-info {
		background-color: white;
		border-color: rgb(26,115,232);
		color: rgb(26,115,232);
	}
  .btn-outline-info:hover {
		background-color: rgb(26,115,232);
		border-color: rgb(26,115,232);
		color: white;
	}
	.dropcap {
		padding-left: 5px;
		padding-right: 5px;
		float: left;
		position: relative;
		top: -0.25em;
		margin-bottom: -0.5em;
		margin-left: -6px;
		margin-right: 2px;
		font-size: 45px;
	}
	.google-maps {
		position: relative;
		padding-bottom: 60%;
		height: 0;
		overflow: hidden;
	}
	.google-maps iframe {
		position: absolute;
		top: 0;
		left: 0;
		width: 100% !important;
		height: 100% !important;
	}
	</style>
</head>

<body>

<div class="container">




<nav class="navbar navbar-expand-lg navbar-light bg-white border-bottom sticky-top">
  <a class="navbar-brand" href="/">CHARLES ROCABERT <span class="sr-only">(current)</span></a>
  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavDropdown" aria-controls="navbarNavDropdown" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon"></span>
  </button>
  <div class="collapse navbar-collapse" id="navbarNavDropdown">
    <ul class="navbar-nav">
      
      <li class="nav-item ">
        <a class="nav-link" href="/">Home</a>
      </li>
      
      <li class="nav-item active">
        <a class="nav-link" href="/research/">Research</a>
      </li>
      
      <li class="nav-item ">
        <a class="nav-link" href="/software/">Software</a>
      </li>
      
      <li class="nav-item ">
        <a class="nav-link" href="/publications/">Publications</a>
      </li>
      
      <li class="nav-item ">
        <a class="nav-link" href="/cv/">CV</a>
      </li>
      
    </ul>
  </div>
  <div class="collapse navbar-collapse" id="navbarNavDropdown">
    <ul class="navbar-nav ml-auto">
      <li class="nav-item" style="margin-right:10px;">
        <a href="https://github.com/charlesrocabert" target="_blank" class="text-dark"><i class="fab fa-github fa-2x"></i></a>&nbsp;&nbsp;
      </li>
      <li class="nav-item" style="margin-right:10px;">
        <a href="https://www.researchgate.net/profile/Charles_Rocabert" target="_blank" class="text-dark"><i class="fab fa-researchgate fa-2x"></i></a>&nbsp;&nbsp;
      </li>
      <li class="nav-item" style="margin-right:10px;">
        <a href="https://www.linkedin.com/in/charles-rocabert/" target="_blank" class="text-dark"><i class="fab fa-linkedin-in fa-2x"></i></a>&nbsp;&nbsp;
      </li>
      <li class="nav-item" style="margin-right:10px;">
        <a href="https://twitter.com/CharlesRocabert" target="_blank" class="text-dark"><i class="fab fa-twitter fa-2x"></i></a>&nbsp;&nbsp;
      </li>
      <li class="nav-item" style="margin-right:10px;">
        <a href="mailto:charles[dot]rocabert[at]helsinki[dot]fi" target="_blank" class="text-dark"><i class="fas fa-envelope fa-2x"></i></a>
      </li>
    </ul>
  </div>
</nav>



<div class="row mt-3 mb-5">
  <div class="col">

    <h4 class="text-secondary font-weight-light"><img src="../img/erlenmeyer.png" alt="Erlenmeyer" width=40px> RESEARCH PROJECTS</h4>

    <!-- Research topic 6 -->

    <!-- Research topic 5 -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <img src="../img/project1.png" alt="Project 1" width=120px class="img-fluid">
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">

              </div>
              <br/><br/>
              <h5><span class="badge badge-pill badge-success">ongoing project</span><br/></h5>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> Deciphering the genetic architecture of polygenic adaptation in <em>Tribolium castaneum</em></h5>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">M</span>any adaptation processes have a polygenic genetic basis. Characterizing the dynamics of polygenic adaptation in evolving populations and their underlying genetic architecture is difficult, and requires the development of new experimental approaches and statistical methods. In particular, the detection of genetic markers associated to polygenic traits is challenging, as weak but meaningful associations are often under the barrier of statistical detection, in addition to linkage desequilibrium and other known biases (<a href="https://doi.org/10.1146/annurev-genom-083115-022316" target="_blank">Sella & Barton, 2019</a>).</p>
                <p class="card-text text-justify font-weight-light">In this context, experimental evolution is a promising tool. By creating the conditions of polygenic adaptation in a controlled environment, this approach allows to characterize the adaptive dynamics at all the levels of the genotype-to-phenotype map, at the resolution of individuals.</p>
                <p class="card-text text-justify font-weight-light">During this project, I will use the data collected during an experimental evolution with <em>Tribolium castaneum</em> (the red floor beetle, see <em>e.g.</em> <a href="https://doi.org/10.1371/journal.pgen.1008768" target="_blank">Koch & Guillaume, 2020</a>) to decipher the genetic architecture of polygenic adaptation. The objective of this project is to better understand the adaptation of populations to environmental change, as well as offering new methods to detect genetic markers of polygenic adaptation.</p>
                <p class="card-text text-justify font-weight-light">Collaborators (by alphabetical order): Frédéric Guillaume, Eva L. Koch.</p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- Research topic 4 -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <img src="../img/project2.png" alt="Project 2" width=200px class="img-fluid">
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
              <a href="https://www.biorxiv.org/content/10.1101/2022.08.15.503737v1" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fas fa-pen-nib"></i> Pre-print (BioRXiv, 2022)</a>
              <a href="https://github.com/charlesrocabert/MetEvolSim" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> MetEvolSim repository</a>
              <a href="https://pypi.org/project/MetEvolSim/" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-python"></i> Pypi package</a>
              </div>
              <br/><br/>
              <h5><span class="badge badge-pill badge-success">ongoing project</span><br/></h5>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> Modeling the principles behind metabolome conservation in mammals</h5>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">T</span>raditionally, metabolic evolution has been primarily studied by comparing enzyme content and pathway structure across species. Recent progress in metabolomics technology allows the systematic quantification of metabolite concentrations, the metabolome, offering a new perspective to interrogate the evolution of metabolic systems.</p>
                <p class="card-text text-justify font-weight-light">Within this interdisciplinary project, I explored the evolving properties of kinetic models of mammalian metabolism using computer simulations. I developed <a href="https://github.com/charlesrocabert/MetEvolSim" target="_blank">MetEvolSim</a> (Metabolome Evolution Simulator), a simulation framework publicly available as a user friendly <a href="https://pypi.org/project/MetEvolSim/" target="_blank">Python package</a>. MetEvolSim takes as an input any SBML metabolic network model, as soon as the kinetic model is fully specified, and a stable steady-state exists. Steady-state concentrations are computed thanks to Copasi software. Using this model, I have shown that abundant metabolites are more coupled to fitness, and are therefore more conserved through evolution <strong>(<a href="https://www.biorxiv.org/content/10.1101/2022.08.15.503737v1" target="_blank">one pre-print in BioRXiv</a>)</strong>.</p>
                <p class="card-text text-justify font-weight-light">Collaborators (by alphabetical order): Gábor Boross, Orsolya Liska, Balázs Papp.</p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- Research topic 3 -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <img src="../img/project3.png" alt="Project 3" width=250px class="img-fluid">
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
              <a href="https://doi.org/10.1111/evo.14083" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fas fa-pen-nib"></i> Publication (Evolution, 2020)</a>
              <br/>
              <a href="https://hal.archives-ouvertes.fr/hal-02402443/document" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fas fa-scroll"></i> Poster (EvoLyon, 2019)</a>
              <br />
              <a href="https://github.com/charlesrocabert/SigmaFGM" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> &sigma;FGM repository</a>
              </div>
              <br/><br/>
              <h5><span class="badge badge-pill badge-success">ongoing project</span><br/></h5>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> Impact of an evolvable phenotypic noise on the adaptation rate of complex organisms</h5>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">H</span>ow does an evolvable phenotypic noise (the variability of isogenic populations in constant environment) impact the probability to fix beneficial mutations? How do the shape of the fitness landscape and the complexity of the phenotype relate to this mechanism? I asked these questions in <a href="https://doi.org/10.1111/evo.14083" target="_blank">this publication</a>, where I show that phenotypic noise promotes the fixation of beneficial mutations on plateauing regions of the fitness landscape, and compensates for the cost of complexity. In particular, I show that the Gaussian fitness function classicaly used in quantitative genetics is a non-generic case regarding the effect of phenotypic noise on the gradient of selection, suggesting that other forms of fitness function (even close to a Gaussian shape) are preferable for who wants to study phenotypic noise.</p>
                <p class="card-text text-justify font-weight-light">These results raise many questions, <em>e.g.</em> in predictive evolution. I am always happy to <a href="mailto:charles[dot]rocabert[at]helsinki[dot]fi" target="_blank">discuss and share ideas</a> on this topic.</p>
                <p class="card-text text-justify font-weight-light">For a shorter take-home message, check out this <a data-toggle="collapse" href="#collapse" aria-expanded="false" aria-controls="collapse">drawing</a> by P. Biller.</p>
                <div class="collapse" id="collapse">
                    <img src="../img/a_long_time_ago.png" alt="A long time ago..." class="img-fluid">
                </div>
                <p class="card-text text-justify font-weight-light">Collaborators (by alphabetical order): Samuel Bernard, Guillaume Beslon, Carole Knibbe.</p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- Research topic 2 -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <img src="../img/project4.png" alt="Project 4" width=200px class="img-fluid">
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
              <a href="https://github.com/charlesrocabert/MoRIS" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> MoRIS repository</a>
              </div>
              <br/><br/>
              <h5><span class="badge badge-pill badge-success">ongoing project</span><br/></h5>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> Integrating road network topology for modeling human-mediated dispersal in terrestrial landscapes</h5>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">B</span>iological invasions are a major threat for biodiversity and human societies. Human-mediated dispersal (HMD) is responsible for transporting invasive species spread through landscapes. Existing models do not account for the topology of the road network, despite its important role in spatially constraining human travels.</p>

                <p class="card-text text-justify font-weight-light">In collaboration with Jérôme M.W. Gippet and Serge Fenet, I developed <a href="https://github.com/charlesrocabert/MoRIS" target="_blank">MoRIS</a> (Model of Routes of Invasive Spread), a spatially explicit and mechanistic stochastic jump model to simulate HMD in terrestrial landscapes. Our modeling approach allows to easily account for the influence of the road network topology on the direction of human travels. By coupling the model to a data-based optimization algorithm (CMA-ES), we have shown that integrating the topology of the road network strongly improve our ability to explain the current distribution of the invasive ant <em>Lasius neglectus</em> in the middle Rhône valley, France <strong>(one manuscript in preparation)</strong>.</p>

                <p class="card-text text-justify font-weight-light">Collaborators (by alphabetical order): Aymeric Bonnamour, Serge Fenet, Jérôme M.W. Gippet, Bernard Kaufmann.</p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- Research topic 1 -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <img src="../img/project5.png" alt="Project 5" width=250px class="img-fluid">
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
              <a href="https://doi.org/10.1371/journal.pcbi.1005459
  " target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fas fa-pen-nib"></i> Publication (PLoS CB, 2017)</a>
              <br/>
              <a href="https://github.com/charlesrocabert/Thesis/blob/master/phd_manuscript.pdf
  " target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fas fa-graduation-cap"></i> PhD manuscript (Nov. 2017)</a>
              <br/>
              <a href="https://github.com/charlesrocabert/Evo2Sim" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> Evo<sup>2</sup>Sim repository</a>
              <br/>
              <a href="www.evovo.eu" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fas fa-link"></i> EvoEvo project website</a>
              </div>
              <br/><br/>
              <h5><span class="badge badge-pill badge-secondary">finished project</span></h5>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> Modeling the evolution of bacterial micro-organisms with numerical simulations</h5>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">E</span>volution is the major source of complexity on Earth, at the origin of all the species we can observe, interact with or breed. On a smaller scale, evolution is at the heart of the adaptation process for many species, in particular micro-organisms (<em>e.g.</em> bacteria, viruses...). These organisms are not only organized by evolution, they are also organized to evolve (<em>i.e.</em> "<a href="https://evoevo.liris.cnrs.fr/what-is-evoevo/" target="_blank">evolution evolves</a>").</p>

                <p class="card-text text-justify font-weight-light">During my PhD, I joined the <a href="https://evoevo.liris.cnrs.fr/" target="_blank">EvoEvo European project</a> to create an integrated computational model of evolution. I developed <a href="https://github.com/charlesrocabert/Evo2Sim" target="_blank">Evo<sup>2</sup>Sim</a> (Evolution of Evolution Simulator), a multi-scale evolutionary model which takes into account the (ultra-fast) dynamics of metabolic networks, the (fast) dynamics of gene regulatory networks, the (medium to slow) dynamics of resources in the ecosystem and the (slow) evolutionary dynamics of genes and genome structure. Evo<sup>2</sup>Sim is designed to perform <em>in silico</em> experimental evolution experiments, that mimick real experimental evolution protocols. A detailed description of the model and its background can be found in my <a href="https://github.com/charlesrocabert/Thesis/blob/master/phd_manuscript.pdf
    " target="_blank">PhD manuscript</a>.</p>

                <p class="card-text text-justify font-weight-light">Using this model, I have contributed to understand why stable diversification events by cross-feeding emerge in Lenski's <em>E. coli</em> long term evolution experiment (<a href="http://myxo.css.msu.edu/ecoli/" target="_blank">LTEE</a>), due to the seasonality introduced by the daily tranfers (see <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005459" target="_blank">my publication in PLoS Computational Biology</a>).</p>

                <p class="card-text text-justify font-weight-light">Collaborators (by alphabetical order): Guillaume Beslon, Jessika Consuegra, Carole Knibbe, Dominique Schneider.</p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

  </div>
</div>


<div class="row">
  <div class="col text-center border-top">
    <small class="text-muted"> &copy; 2020-2023 Charles Rocabert. All rights reserved. Powered by Hugo, Staticman and Github Pages. Drawings: &copy; Manuela Michel Dikongué.</small>
  </div>
</div>

</div>



<script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.6/umd/popper.min.js" integrity="sha384-wHAiFfRlMFy6i5SRaxvfOCifBUQy1xHdJ/yoi7FRNXMRBu5WHdZYu1hA6ZOblgut" crossorigin="anonymous"></script>
<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/js/bootstrap.min.js" integrity="sha384-B0UglyR+jN6CkvvICOB2joaf5I4l3gm9GU6Hc1og6Ls7i6U/mkkaduKaBhlAXv9k" crossorigin="anonymous"></script>

</body>

</html>

