<!DOCTYPE html>

<html lang="en">

<head>
  
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-R5FPWGD7WK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-R5FPWGD7WK');
  </script>
	
	<title>Charles Rocabert | Software</title>
	
	<link rel="apple-touch-icon" sizes="180x180" href="img/favicon.png">
	<link rel="icon" type="image/png" sizes="32x32" href="img/favicon.png">
	<link rel="icon" type="image/png" sizes="16x16" href="img/favicon.png">
	
	<meta charset="utf-8" />
	<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
	
  <link rel="stylesheet" href="css/bootstrap-4.2.1.min.css">
  <link rel="stylesheet" href="../css/bootstrap-4.2.1.min.css">
  <link rel="stylesheet" href="../../css/bootstrap-4.2.1.min.css">
	<link rel="stylesheet" href="css/fontawesome-5.6.3.min.css">
  <link rel="stylesheet" href="../css/fontawesome-5.6.3.min.css">
  <link rel="stylesheet" href="../../css/fontawesome-5.6.3.min.css">

  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css" integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS" crossorigin="anonymous">
	<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">

  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.2/dist/katex.min.css" integrity="sha384-MlJdn/WNKDGXveldHDdyRP1R4CTHr3FeuDNfhsLPYrq2t0UBkUdK2jyTnXPEK1NQ" crossorigin="anonymous">

  
  <script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.2/dist/katex.min.js" integrity="sha384-VQ8d8WVFw0yHhCk5E8I86oOhv48xLpnDZx5T9GogA/Y84DcCKWXDmSDfn13bzFZY" crossorigin="anonymous"></script>

  
  <script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.2/dist/contrib/auto-render.min.js" integrity="sha384-+XBljXPPiv+OzfbB3cVmLHf4hdUFHlWNZN5spNQ7rmHTXpd7WvJum6fIACpNNfIR" crossorigin="anonymous"
      onload="renderMathInElement(document.body);"></script>
  <script>
  document.addEventListener("DOMContentLoaded", function()
  {
    renderMathInElement(document.body,
    {
      delimiters: [
        {left: "$$", right: "$$", display: true},
        {left: "$", right: "$", display: false}
      ]
    });
  });
  </script>

	<style>
	html {
		font-size: 1rem;
	}
	a {
		color: rgb(64,139,179);
		font-weight: normal;
	}
	.active a {
		color: rgb(26,115,232) !important;
		font-weight: bold;
		border-bottom: 5px solid !important;
		border-bottom-color: rgb(26,115,232);
	}
	.btn-outline-info {
		background-color: white;
		border-color: rgb(26,115,232);
		color: rgb(26,115,232);
	}
  .btn-outline-info:hover {
		background-color: rgb(26,115,232);
		border-color: rgb(26,115,232);
		color: white;
	}
	.dropcap {
		padding-left: 5px;
		padding-right: 5px;
		float: left;
		position: relative;
		top: -0.25em;
		margin-bottom: -0.5em;
		margin-left: -6px;
		margin-right: 2px;
		font-size: 45px;
	}
	.google-maps {
		position: relative;
		padding-bottom: 60%;
		height: 0;
		overflow: hidden;
	}
	.google-maps iframe {
		position: absolute;
		top: 0;
		left: 0;
		width: 100% !important;
		height: 100% !important;
	}
	</style>
</head>

<body>

<div class="container">




<nav class="navbar navbar-expand-lg navbar-light bg-white border-bottom sticky-top">
  <a class="navbar-brand" href="/">CHARLES ROCABERT <span class="sr-only">(current)</span></a>
  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavDropdown" aria-controls="navbarNavDropdown" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon"></span>
  </button>
  <div class="collapse navbar-collapse" id="navbarNavDropdown">
    <ul class="navbar-nav">
      
      <li class="nav-item ">
        <a class="nav-link" href="/">Home</a>
      </li>
      
      <li class="nav-item ">
        <a class="nav-link" href="/research/">Research</a>
      </li>
      
      <li class="nav-item active">
        <a class="nav-link" href="/software/">Software</a>
      </li>
      
      <li class="nav-item ">
        <a class="nav-link" href="/publications/">Publications</a>
      </li>
      
      <li class="nav-item ">
        <a class="nav-link" href="/cv/">CV</a>
      </li>
      
    </ul>
  </div>
  <div class="collapse navbar-collapse" id="navbarNavDropdown">
    <ul class="navbar-nav ml-auto">
      <li class="nav-item" style="margin-right:10px;">
        <a href="https://github.com/charlesrocabert" target="_blank" class="text-dark"><i class="fab fa-github fa-2x"></i></a>&nbsp;&nbsp;
      </li>
      <li class="nav-item" style="margin-right:10px;">
        <a href="https://www.researchgate.net/profile/Charles_Rocabert" target="_blank" class="text-dark"><i class="fab fa-researchgate fa-2x"></i></a>&nbsp;&nbsp;
      </li>
      <li class="nav-item" style="margin-right:10px;">
        <a href="https://www.linkedin.com/in/charles-rocabert/" target="_blank" class="text-dark"><i class="fab fa-linkedin-in fa-2x"></i></a>&nbsp;&nbsp;
      </li>
      <li class="nav-item" style="margin-right:10px;">
        <a href="https://twitter.com/CharlesRocabert" target="_blank" class="text-dark"><i class="fab fa-twitter fa-2x"></i></a>&nbsp;&nbsp;
      </li>
      <li class="nav-item" style="margin-right:10px;">
        <a href="mailto:charles[dot]rocabert[at]hhu[dot]de" target="_blank" class="text-dark"><i class="fas fa-envelope fa-2x"></i></a>
      </li>
    </ul>
  </div>
</nav>



<div class="row mt-3 mb-5">
  <div class="col">

    <h4 class="text-secondary font-weight-light"><img src="../img/piece_of_code.png" alt="Piece of code" width=60px> SOFTWARE</h4>

    <!-- Navbar -->
    <br/>
    <ul class="nav nav-tabs">
      <li class="nav-item">
        <a class="nav-link" href="#metevolsim"><i class="fas fa-caret-down"></i> MetEvolSim</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" href="#moris"><i class="fas fa-caret-down"></i> MoRIS</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" href="#multiomics_pipeline"><i class="fas fa-caret-down"></i> Multi-omics pipeline</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" href="#sigmafgm"><i class="fas fa-caret-down"></i> &sigma;FGM</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" href="#evo2sim"><i class="fas fa-caret-down"></i> Evo<sup>2</sup>Sim</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" href="#puutools"><i class="fas fa-caret-down"></i> puutools</a>
      </li>
    </ul>
    <br/>

    <!-- MetEvolSim -->
    <div class="row justify-content-center mt-4" name="metevolsim">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <a name="metevolsim"></a>
              <img src="../img/metevolsim_logo.png" alt="MetEvolSim logo" width=200px class="img-fluid">
              <br/><br />
              <span class="text-muted">A Python package to simulate the long-term evolution of metabolic levels.</span>
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
              <a href="https://github.com/charlesrocabert/MetEvolSim" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> MetEvolSim repository</a>
              <a href="https://pypi.org/project/MetEvolSim/" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-python"></i> Pypi package</a>
              </div>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> MetEvolSim (Metabolome Evolution Simulator)</h5>
                <p align="center">
                  <a href="https://badge.fury.io/py/MetEvolSim"><img src="https://badge.fury.io/py/MetEvolSim.svg" alt="PyPI version"></a>&nbsp;
                  <a href="https://github.com/charlesrocabert/MetEvolSim/actions" target="_blank"><img src="https://github.com/charlesrocabert/MetEvolSim/workflows/Upload Python Package/badge.svg" /></a>&nbsp;
                  <a href="https://github.com/charlesrocabert/MetEvolSim/LICENSE.html" target="_blank"><img src="https://img.shields.io/badge/License-GPLv3-blue.svg" /></a>
                </p>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">M</span>etEvolSim (<em>Metabolome Evolution Simulator</em>) is a Python package providing numerical tools to simulate the long-term evolution of metabolic levels in kinetic models of metabolism.
                MetEvolSim takes as an input any <a href="http://sbml.org/Main_Page" target="_blank">SBML</a> metabolic network model, as soon as the kinetic model is fully specified, and a stable steady-state exists. Steady-state concentrations are computed thanks to <a href="http://copasi.org/" target="_blank">Copasi</a> software.</p>
                <p class="card-text text-justify font-weight-light">MetEvolSim is distributed as a user friendly <a href="https://pypi.org/project/MetEvolSim/" target="_blank">Pypi package</a>. The full description of the simulation framework, first usage and some examples are provided on the <a href="https://pypi.python.org/pypi/MetEvolSim/" target="_blank">Github page</a>.</p>
                <h5>Publications and communications</h5>
                <p class="card-text text-justify font-weight-light">
                • <strong>2023:</strong> <em>Proceedings of the National Academy of Sciences.</em> 120 (35), e2302147120. <a href="https://doi.org/10.1073/pnas.2302147120" target="_blank">(link)</a>.
                <br/>
                • <strong>Feb. 2020:</strong> <em>Evolutionary Systems Biology, Wellcome Genome Campus.</em> (Cambridge, United Kingdom) <a href="https://coursesandconferences.wellcomegenomecampus.org/wp-content/uploads/2020/01/Evolutionary-Systems-Biology-abstract.pdf" target="_blank">(link)</a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- MoRIS -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <a name="moris"></a>
              <img src="../img/moris_logo.png" alt="MoRIS logo" width=200px class="img-fluid">
              <br/><br />
              <span class="text-muted">A simulator of human-mediated dispersal via transportation networks.</span>
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
              <a href="https://github.com/charlesrocabert/MoRIS" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> MoRIS repository</a>
              <a href="https://github.com/charlesrocabert/MoRIS/blob/master/INPUT_FILES_TUTORIAL.md" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fas fa-link"></i> Input file tutorial</a>
            </div>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> MoRIS (Model of Routes of Invasive Spread)</h5>
                <p align="center">
                  <a href="https://github.com/charlesrocabert/MoRIS/releases/latest" target="_blank"><img src="https://img.shields.io/github/release/charlesrocabert/MoRIS/all.svg" /></a>&nbsp;
                  <a href="https://github.com/charlesrocabert/MoRIS/actions" target="_blank"><img src="https://github.com/charlesrocabert/MoRIS/workflows/CMake/badge.svg" /></a>&nbsp;
                  <a href="https://github.com/charlesrocabert/MoRIS/LICENSE.html" target="_blank"><img src="https://img.shields.io/badge/License-GPLv3-blue.svg" /></a>
                </p>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">M</span>oRIS (<em>Model of Routes of Invasive Spread</em>) is a spatially explicit spread model designed to simulate invasive species dispersal by transport at local to regional spatial scales. MoRIS is an innovative, yet simple model, taking into account the road network topology to influence the direction of dispersal events. MoRIS is designed to minimize <em>a priori</em> making (<em>e.g.</em>, expert knowledge), to enable the estimation of human-mediated dispersal parameters based on a simple presence/absence locations dataset and to produce predictive maps of spread.</p>
                <p class="card-text text-justify font-weight-light">To learn how to use MoRIS, first install the <a href="https://github.com/charlesrocabert/MoRIS/releases/latest" target="_blank">latest release</a>, and follow the <a href="https://github.com/charlesrocabert/MoRIS/" target="_blank">installation instructions and the first usage guide</a>. MoRIS software requires input files built in a certain way. To learn how to build this files, read the <a href="https://github.com/charlesrocabert/MoRIS/blob/master/INPUT_FILES_TUTORIAL.md" target="_blank">input files tutorial</a>.</p>
                <h5>Publications and communications</h5>
                <p class="card-text text-justify font-weight-light">
                • <strong>2023:</strong> <em>Ecography.</em> in press. <a href="https://doi.org/10.1111/ecog.07068" target="_blank">(link)</a>
                <br/>
                • <strong>Aug. 2016:</strong> <em>Proceedings of IENE 2016 conference. 5th International Conference on Ecology and Transportation: Integrating Transport Infrastructures with Living Landscapes</em> (Lyon, France). <a href="http://hal.cirad.fr/LJK_MAD_STEEP/hal-01412280v1" target="_blank">(link)</a>
                <br/>
                • <strong>Jul. 2015:</strong> <em>Proceedings of World Conference on Natural Resource Modeling</em> (Bordeaux, France). <a href="https://hal.archives-ouvertes.fr/hal-01242828/" target="_blank">(link)</a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- Multi-omics pipeline -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <a name="multiomics_pipeline"></a>
              <img src="../img/Tcastaneum_logo.png" alt="Tcastaneum logo" width=200px class="img-fluid">
              <br/><br />
              <span class="text-muted">Multi-omics pipeline to detect signals of selection during stressful environmental change.</span>
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
                <a href="https://github.com/charlesrocabert/Tribolium-castaneum-transcriptomics-pipeline" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> Pipeline repository</a>
              </div>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> <em>Tribolium castaneum</em> multi-omics pipeline</h5>
                <p align="center">
                  <a href="https://github.com/charlesrocabert/Tribolium-castaneum-transcriptomics-pipeline/LICENSE.html" target="_blank"><img src="https://img.shields.io/badge/License-GPLv3-blue.svg" /></a>
                </p>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">T</span>he objective of this pipeline is to associate signals of selection at the genomic and gene expression levels. The originality of this approach is to both detect allele frequency changes (AFCs) and gene expression changes through generations, using the same individual-level RNA-seq dataset. Recent studies have demonstrated the possibility to detect SNPs from RNA-seq data, assuming that only expressed regions of the genome are available. The approach covers a vast amount of methodologies, including variants call, the calculation of AFCs and the genetic map, read counts or the detection of gene expression modules..</p>
                <h5>Publications and communications</h5>
                <p class="card-text text-justify font-weight-light">
                • One manuscript in preparation.
                <br/>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- SigmaFGM -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <a name="sigmafgm"></a>
              <img src="../img/sigmaFGM_logo.png" alt="sigmaFGM logo" width=200px class="img-fluid">
              <br/><br />
              <span class="text-muted">Individual-based implementation of Fisher's geometric model with evolvable phenotypic noise.</span>
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
                <a href="https://github.com/charlesrocabert/SigmaFGM" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> &sigma;FGM repository</a>
              </div>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> &sigma;FGM (Fisher's geometric model with evolvable phenotypic noise)</h5>
                <p align="center">
                  <a href="https://github.com/charlesrocabert/SigmaFGM/releases/latest" target="_blank"><img src="https://img.shields.io/github/release/charlesrocabert/SigmaFGM/all.svg" /></a>&nbsp;
                  <a href="https://github.com/charlesrocabert/SigmaFGM/actions" target="_blank"><img src="https://github.com/charlesrocabert/SigmaFGM/workflows/CMake/badge.svg" /></a>&nbsp;
                  <a href="https://github.com/charlesrocabert/SigmaFGM/LICENSE.html" target="_blank"><img src="https://img.shields.io/badge/License-GPLv3-blue.svg" /></a>
                </p>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">&sigma;</span>FGM simulates adaptive evolution in Fisher's geometric model with an evolvable phenotypic noise. A population of individuals is placed under stabilizing selection and must evolve towards a fitness optimum. The fitness function is configurable and can adopt non-Gaussian shapes. Phenotypic noise is modeled by an evolvable multivariate normal distribution. Simulations are fully configurable. See <a href="https://github.com/charlesrocabert/SigmaFGM/" target="_blank">the installation instructions and a first usage</a>, and <a href="https://doi.org/10.1111/evo.14083" target="_blank">Rocabert et al. (2020)</a> for a full description of the underlying mathematical model.</p>
                <h5>Publications and communications</h5>
                <p class="card-text text-justify font-weight-light">
                • <strong>2020:</strong> <em>Evolution</em> 74 (10), 2221-2237. <a href="https://doi.org/10.1111/evo.14083" target="_blank">(link)</a>
                <br/>
                • <strong>Nov. 2019:</strong> <em>Conference EvoLyon 2019</em>. <a href="https://hal.archives-ouvertes.fr/hal-02402443/file/poster2_v3.pdf" target="_blank">(link)</a>
                <br/>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- Evo2Sim -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <a name="evo2sim"></a>
              <img src="../img/evo2sim_logo.png" alt="Evo2Sim logo" width=200px class="img-fluid">
              <br/><br />
              <span class="text-muted">Individual-based and multi-scale evolutionary model, dedicated to <em>in silico</em> experimental evolution.</span>
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
                <a href="https://github.com/charlesrocabert/Evo2Sim" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> Evo<sup>2</sup>Sim repository</a>
                <a href="http://www.evoevo.eu/" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fas fa-link"></i> EvoEvo project website</a>
                <a href="../doc/evo2sim_simulation_example/viewer/viewer.html" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="far fa-window-maximize"></i> Simulation example</a>
              </div>
            </div>
            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> Evo<sup>2</sup>Sim (Evolution of Evolution Simulator)</h5>
                <p align="center">
                  <a href="https://github.com/charlesrocabert/Evo2Sim/releases/latest" target="_blank"><img src="https://img.shields.io/badge/version- v1.0.2-green.svg" /></a>&nbsp;
                  <a href="https://github.com/charlesrocabert/Evo2Sim/releases/latest" target="_blank"><img src="https://img.shields.io/badge/build-passing-green.svg" /></a>&nbsp;
                  <a href="https://www.gnu.org/licenses/gpl-3.0" target="_blank"><img src="https://img.shields.io/badge/license-GPL v3-blue.svg" /></a>
                </p>
                <p class="card-text text-justify font-weight-light"><span class="dropcap">E</span>vo<sup>2</sup>Sim (<em>Evolution of Evolution Simulator</em>) is a multi-scale evolutionary model which takes into account the (ultra-fast) dynamics of metabolic networks, the (fast) dynamics of gene regulatory networks, the (medium to slow) dynamics of resources in the ecosystem and the (slow) evolutionary dynamics of genes and genome structure. Evo<sup>2</sup>Sim is designed to perform <em>In Silico</em> Experimental Evolution (ISEE) experiments, that mimick real experimental evolution protocols.</p>
                <p class="card-text text-justify font-weight-light">The software is equipped with the whole tool case of experimental setups, competition assays, phylogenetic analysis. Simulations can be analyzed thanks to the HTML viewer (see <a href="../doc/evo2sim_simulation_example/viewer/viewer.html" target="_blank">the example</a>), tracking and displaying on the fly every simulation events, from the phylogeny to ecological interactions.</p>
                <p class="card-text text-justify font-weight-light">Evo<sup>2</sup>Sim was first developed during the EvoEvo project (<a href="http://www.evoevo.eu/" target="_blank">www.evoevo.eu</a>), funded by the European Commission (FP7-ICT-610427, FET Proactive: Evolving Living Technologies).</p>
                <h5>Publications and communications</h5>
                <p class="card-text text-justify font-weight-light">
                • <strong>2017:</strong> <em>PLoS computational biology</em>, 13(3), e1005459. <a href="https://doi.org/10.1371/journal.pcbi.1005459" target="_blank">(link)</a>
                <br/>
                • <strong>Sep. 2017:</strong> <em>Proceedings of the 14th European Conference on Artificial Life (ECAL)</em> (Villeurbanne, France). <a href="https://hal.archives-ouvertes.fr/hal-01569093/" target="_blank">(link)</a>
                <br/>
                • <strong>Jul. 2015:</strong> <em>Proceedings of the EvoEvo Workshop, Satellite workshop of ECAL 2015</em> (York, UK). <a href="https://hal.inria.fr/hal-01252796/" target="_blank">(link)</a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- puutools -->
    <div class="row justify-content-center mt-4">
      <div class="col-lg-11">
        <div class="shadow-sm card border-muted">
          <div class="row no-gutters">
            <div class="col-md-4 p-4 text-center">
              <a name="puutools"></a>
              <img src="../img/puutools_logo.png" alt="MoRIS logo" width=200px class="img-fluid">
              <br/><br />
              <span class="text-muted">Easy-to-use C++ library for the live tracking of lineage and coalescence trees in individual-based forward-in-time simulations.</span>
              <br /><br />
              <div class="btn-group-vertical btn-block" role="group" aria-label="Button group with nested dropdown">
              <a href="https://github.com/charlesrocabert/puutools" target="_blank" type="button" class="text-left btn btn-outline-info"><i class="fab fa-github"></i> puutools repository</a>
              </div>
            </div>

            <div class="col-md-8">
              <div class="card-body">
                <h5 class="card-title"><i class="fas fa-caret-down"></i> puutools (live tracking of lineage and coalescence trees)</h5>
                <p align="center">
                  <a href="https://github.com/charlesrocabert/puutools/releases/latest"><img src="https://img.shields.io/github/release/charlesrocabert/puutools/all.svg" /></a>&nbsp;
                  <a href="https://github.com/charlesrocabert/puutools/actions"><img src="https://github.com/charlesrocabert/puutools/workflows/CMake/badge.svg" /></a>&nbsp;
                  <a href="https://github.com/charlesrocabert/puutools/LICENSE.html"><img src="https://img.shields.io/badge/License-GPLv3-blue.svg" /></a>
                </p>

                <p class="card-text text-justify font-weight-light"><span class="dropcap">P</span>uutools is a C++ library which allows to dynamically create, track and manage lineage and/or coalescence trees during individual-based forward-in-time simulations of evolution.</p>

                <p class="card-text text-justify font-weight-light">With puutools, it is possible to update the lineage and/or the coalescence tree of a population at each generation during a simulation, and to track and recover evolutionary events at any time. puutools is easy to deploy and versatile, providing the user with a small set of integrated functions.</p>

                <p class="card-text text-justify font-weight-light">puutools is an all-in-one library. It does not depend on third-party packages, and its code is kept simple and light. By the way, puutools comes as and header-only library, so specific compilation is not needed.</p>

                <p class="card-text text-justify font-weight-light">The code of puutools has primarily been implemented and optimized during the European project <a href="http://www.evoevo.eu/">EvoEvo</a> (FP7-ICT-610427), as part of the <a href="https://github.com/charlesrocabert/Evo2Sim">Evo<sup>2</sup>Sim</a> software (see <a href="https://doi.org/10.1371/journal.pcbi.1005459">Rocabert et al. 2017</a>). The code has also been used in σFGM framework (see <a href="https://doi.org/10.1111/evo.14083">Rocabert et al. 2020</a>). To date, puutools is distributed as a static library for C++ developers. We plan to provide a Python-binding in a future version.</p>

                <p class="card-text text-justify font-weight-light">You will find a <a href="https://github.com/charlesrocabert/puutools/tree/main/example" target="_blank">complete walk-through example</a> to get used to the main functionalities of puutools.</p>

              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

  </div>
</div>


<div class="row">
  <div class="col text-center border-top">
    <small class="text-muted"> &copy; 2020-2024 Charles Rocabert. All rights reserved. Powered by Hugo and Github Pages. Drawings: &copy; Manuela Michel Dikongué.</small>
  </div>
</div>

</div>



<script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.6/umd/popper.min.js" integrity="sha384-wHAiFfRlMFy6i5SRaxvfOCifBUQy1xHdJ/yoi7FRNXMRBu5WHdZYu1hA6ZOblgut" crossorigin="anonymous"></script>
<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/js/bootstrap.min.js" integrity="sha384-B0UglyR+jN6CkvvICOB2joaf5I4l3gm9GU6Hc1og6Ls7i6U/mkkaduKaBhlAXv9k" crossorigin="anonymous"></script>

</body>

</html>

